### PR TITLE
authz: remove all call sites to dbconn.Global in `PermsSyncer`

### PR DESCRIPTION
**Pull request by unknwon**
This PRs removes two call sites that relies on `dbconn.Global` from `PermsSyncer`, replaced with two new methods in the `PermsStore`.

TODO: Tests will be added after review.

### CODE REVIEW

File: enterprise/cmd/repo-updater/authz/perms_syncer.go [hunk 1]
```diff
@@ -298,17 +291,17 @@ func (s *PermsSyncer) syncRepoPerms(ctx context.Context, repoID api.RepoID) erro
...
-		// Remove existing user from set of pending users
-		delete(bindUsernamesSet, users[i].Username)
+		// Remove existing user from the set of pending users
+		delete(pendingAccountIDsSet, aid)
 	}
 
-	pendingBindUsernames := make([]string, 0, len(bindUsernamesSet))
-	for id := range bindUsernamesSet {
-		pendingBindUsernames = append(pendingBindUsernames, id)
+	pendingAccountIDs := make([]string, 0, len(pendingAccountIDsSet))
```
Comment (tsenart) [inline enterprise/cmd/repo-updater/authz/perms_syncer.go]: Do we have tests for pending? We should also manually test.

Comment (unknwon) [reply enterprise/cmd/repo-updater/authz/perms_syncer.go]: For pending permissions? If so yes:

https://github.com/sourcegraph/sourcegraph/blob/b987679a29fa69fcc30f8a040a87aabaa149ba32/enterprise/cmd/frontend/db/integration_test.go#L30-L31

Comment (tsenart) [reply enterprise/cmd/repo-updater/authz/perms_syncer.go]: I meant the syncer logic here, not the store methods.

Comment (unknwon) [reply enterprise/cmd/repo-updater/authz/perms_syncer.go]: Syncer does not have any tests yet, will ensure this when I add tests (after e2e testing).

**Comment by unknwon** [comment]
Merging as-is, happy to address any post-merge comments!

**Pull request merged**